(MODULES-7857) Support user creation on galera

MaxFedotov opened the pull request:
Support user creation on galera. Add IF NOT EXISTS to CREATE USER in mysql_user provider
Description of issue in jira - https://tickets.puppetlabs.com/browse/MODULES-7857

david22swan commented:
@MaxFedotov Could you add some test coverage for your new feature and update the docs please?

MaxFedotov commented:
@david22swan Hi. Can you please clarify, which docs should be updated? This PR doesn't add new functionality, it's just reuses the same logic, which is already implemented for DROP USER (https://github.com/puppetlabs/puppetlabs-mysql/blob/9ede9a8c3a49ee4ba05ae87ee57073190b8f631a/lib/puppet/provider/mysql_user/mysql.rb#L109). I've looked throught all docs, and wasn't able to find anything related to this.

MaxFedotov commented:
@david22swan I've rebased on current master and slightly modified my PR and make a few fixes for the last pull request you've merged. There were some places, which had different ALTER commands (IF EXISTS was missing) on the same IF clause, and in some places mariadb was missing in IF clause.
Also i've updated tests.
